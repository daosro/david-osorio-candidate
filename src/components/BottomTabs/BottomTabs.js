import React from 'react';

import useStyles from './BottomTabs.style';

const BottomTabs = ({ children }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      {children}
    </div>
  );
};

export default BottomTabs;