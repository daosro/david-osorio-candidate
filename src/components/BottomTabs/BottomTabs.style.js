import { createUseStyles } from 'react-jss';

export default createUseStyles({
  root: {
    display: 'flex',
    backgroundColor: '#202020',
    position: 'sticky',
    bottom: 0,
    height: '50px',
    fontSize: '25px',
    padding: '0 15px',
    alignItems: 'center',
    justifyContent: 'space-between',
    '& a': {
      color: '#ffffff',
    },
    '& a.active': {
      color: '#e1a21e',
    }
  },
}, { name: 'BottomTabs' });