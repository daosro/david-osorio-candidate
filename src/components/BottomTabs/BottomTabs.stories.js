import React from "react";

import {
  MdHome,
  MdLiveTv,
  MdReplay,
  MdViewList,
  MdMenuBook,
} from "react-icons/md";

import BottomTabs from "./BottomTabs";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "NoriginMedia/BottomTabs",
  component: BottomTabs,
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => (
  <BottomTabs {...args}>
    <MdHome />
    <MdLiveTv />
    <MdReplay />
    <MdViewList />
    <MdMenuBook />
  </BottomTabs>
);

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {};
