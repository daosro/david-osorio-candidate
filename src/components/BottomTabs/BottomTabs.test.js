import { render } from "@testing-library/react";
import BottomTabs from "./BottomTabs";

describe("BottomTabs", () => {
  it("should render correctly", () => {
    const { container } = render(<BottomTabs />);
    expect(container).toMatchSnapshot();
  });
  it("should render correctly with children", () => {
    const { container } = render(
      <BottomTabs>
        <div>Mock div</div>
      </BottomTabs>
    );
    expect(container).toMatchSnapshot();
  });
  it("should contain the children", () => {
    const { container } = render(
      <BottomTabs>
        <div>Mock div</div>
      </BottomTabs>
    );
    expect(container.querySelector("div")).toBeInTheDocument();
  });
});
