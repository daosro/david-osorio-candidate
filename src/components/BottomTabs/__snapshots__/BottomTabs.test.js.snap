// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabs should render correctly 1`] = `
<div>
  <div
    class="BottomTabs-root-0-2-1"
  />
</div>
`;

exports[`BottomTabs should render correctly with children 1`] = `
<div>
  <div
    class="BottomTabs-root-0-2-1"
  >
    <div>
      Mock div
    </div>
  </div>
</div>
`;
